File: backend/models/subcomponent.js
import mongoose from "mongoose";

const Schema = mongoose.Schema;

const subcomponentSchema = new Schema(
  {
    name: String,
    weightage: Number,
    totalMarks: Number,
    studentMarks: {
      type: Map,
      of: Number
    },
  }
);

// const marksSchema = new Schema({
//   studentId: { type: Schema.Types.ObjectId, ref: "User" },
//   marks: Number,
// });

const SubComponent = mongoose.model("SubComponent", subcomponentSchema);
//export const Marks = marksSchema;

export default SubComponent;

File: gamification/Gamification/Assets/Scripts/Entities/SubComponent.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class SubComponent
{
    public string Name { get; private set; }
    public int Weightage { get; private set; }
    public int ClassStanding { get; private set; }
    public int TotalMarks { get; private set; }
    public int Marks { get; private set; }

    public SubComponent(string name, int weightage, int classStanding, int totalMarks, int marks)
    {
        Name = name;
        Weightage = weightage;
        ClassStanding = classStanding;
        TotalMarks = totalMarks;
        Marks = marks;
    }

    public Grade CalculateGrade()
    {
        return Grade.A;
    }
}
